mne/io/base.py

```diff
@@ -1542,12 +1546,12 @@ def resample(self, sfreq, npad='auto', window='boxcar', stim_picks=None,
 
             return self
         else:
-            if copy:
-                events = events.copy()
+            # always make a copy of events
+            events = events.copy()
 
             events[:, 0] = np.minimum(
                 np.round(events[:, 0] * ratio).astype(int),
-                self._data.shape[1]
+                self._data.shape[1] + self.first_samp
```

Comment:
The fix

Reply:
Don't you need to subtract the original `raw.first_samp` and the add the modified one at the end?

Reply:
I'm thinking the `np.round(events[:, 0] * ratio).astype(int)` works just fine, and the only problem is the `np.min` (which is there for a corner case, I suppose). This seemed like a too easy fix, but maybe such things do exist...?

Reply:
`np.minimum`

Reply:
Ahh I see, I misread the code. Yeah this looks reasonable to me